### PR TITLE
Eng 1417

### PR DESCRIPTION
**Pull request by jallen2112**
Harbor Integration. Allow pipelines in cluster to push images/charts to harbor or artifactory (Default to harbor). Map harbor/artifactory project credentials to dockercfg.

### CODE REVIEW

File: modules/lead/product/dockercfg.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ 
+  vars = { 
+    email = "jenkins@liatr.io"
+    artifactory_url   = "https://artifactory.toolchain.${var.cluster_domain}/docker-registry/${var.product_name}"
+    artifactory_auth = base64encode(
+      "${data.kubernetes_secret.jenkins_artifactory_credential.data.username}:${data.kubernetes_secret.jenkins_artifactory_credential.data.password}",
+    )   
+    harbor_url = "https://harbor.toolchain.${var.cluster_domain}/${var.product_name}"
+    harbor_auth = base64encode(
+      "robot$imagepusher:${data.kubernetes_secret.product-harbor-creds.data.AUTH}",
```
Comment (JonRudy) [inline modules/lead/product/dockercfg.tf]: Did the idea change of having multiple robot accounts specific for each product or repo?

Comment (jallen2112) [reply modules/lead/product/dockercfg.tf]: No, harbor projects and auth resources are still created per product. The dockercfg contains both artifactory and harbor credentials so pipelines can push to either

Comment (JonRudy) [reply modules/lead/product/dockercfg.tf]: hmm maybe i'm just not understanding the robot fully. I was assuming where you have `imagepusher` you would have to have the name of the specific robot account that gets created for that repo.

Comment (jallen2112) [reply modules/lead/product/dockercfg.tf]: Every product has a robot account named `robot$imagepusher`, which is tied to the product by the specific auth token (robot$imagepusher:AUTHTOKEN). The token for the jordan-lab `robot$imagepusher` robot account won't work for the jon-lab `robot$imagepusher` and vice versa.

Comment (JonRudy) [reply modules/lead/product/dockercfg.tf]: thanks for the explanation

File: modules/lead/product/toolchain.tf [hunk 2]
```diff
@@ -24,6 +24,8 @@ data "template_file" "jenkins_values" {
     appDomain              = "apps.${var.cluster_domain}"
     builder_images_version = var.builder_images_version
     allow_anonymous_read   = var.enable_keycloak ? "false" : "true"
+    jenkins-repository-dockercfg = "jenkins-repository-dockercfg"
```
Comment (mrparkers) [inline modules/lead/product/toolchain.tf]: this should be referenced from `kubernetes_secret.jenkins_repository_dockercfg`, not hardcoded

**Pull request closed without merging**